### PR TITLE
Lock Release

### PR DESCRIPTION
**Pull request by tbwebb22**
This is a very WIP PR to just demonstrate the concept we talked about today. 

I've added some functions & variables to the `LockRelease` from `ERC20Votes` so that delegation can be done within that contract. There are some @todo tags throughout the code highlighting some of the important lines that we need to make sure are correct.

### CODE REVIEW

File: contracts/DCNTToken.sol [hunk 1]
```diff
@@ -15,9 +16,10 @@ contract DCNTToken is ERC20VotesLockable {
     error MintTooSoon();
 
     /// @param _supply amount of tokens to mint at Token Generation Event
-    constructor(uint256 _supply) ERC20("Decent", "DCNT") ERC20Permit("Decent") {
+    constructor(uint256 _supply, address _owner) ERC20("Decent", "DCNT") ERC20Permit("Decent") {
```
Comment (herbig) [inline contracts/DCNTToken.sol:19]: you could do `Ownable(_owner)` here rather than the transfer line below

Comment (herbig) [reply contracts/DCNTToken.sol:19]: or just Ownable(msg.sender)

File: contracts/LockRelease.sol [hunk 2]
```diff
@@ -69,6 +88,10 @@ contract LockRelease is ILockRelease {
 
             Schedule memory schedule = Schedule(amount, 0);
             schedules[beneficiary] = schedule;
+
+            // @todo: Ensure the following lines are correct
+            _delegate(beneficiary, beneficiary);
+            _moveVotingPower(address(0), beneficiary, amount);
```
Comment (herbig) [inline contracts/LockRelease.sol:94]: shouldn't the `_delegate` function above this line be moving the voting power as well?

File: contracts/DCNTToken.sol (hunk 1 above)
Comment (tbwebb22) [reply contracts/DCNTToken.sol:19]: Also, I'd assume we'd want the ability to initialize the owner address to one different than what its deployed from? So we could initialize owner to be the DAO address

Comment (herbig) [reply contracts/DCNTToken.sol:19]: ah cool cool, makes sense

File: contracts/LockRelease.sol (hunk 2 above)
Comment (tbwebb22) [reply contracts/LockRelease.sol:94]: hmm so when `_delegate(beneficiary, beneficary) is called, the `delegatorBalance` in the code snippet below (from `ERC20Votes.sol` will be 0, so the `moveVotingPower` that is called from within that function won't actually move any voting power

image

Comment (herbig) [reply contracts/LockRelease.sol:94]: ahhh ok ok, yeah makes sense

**Pull request closed without merging**